src/charm.py

```diff
@@ -602,7 +603,7 @@ def _on_config_changed(self, event) -> None:
                 )
                 return
 
-    def enable_disable_extensions(self, database: str = None) -> None:
+    def enable_disable_extensions(self, database: str = None) -> None:  # noqa: C901
```

Comment:
I know cyclomatic complexity is annoying, but we should either bump the limit or fit within it, not disable it on individual functions.